core/domain/suggestion_services.py

```diff
@@ -1434,42 +1437,83 @@ def _update_suggestion_counts_in_community_contribution_stats(
         suggestions, amount)
 
 
-def update_translation_suggestion(suggestion_id, translation_html):
+def update_translation_suggestion(suggestion_id, translation_html, user_id):
     """Updates the translation_html of a suggestion with the given
     suggestion_id.
 
     Args:
         suggestion_id: str. The id of the suggestion to be updated.
         translation_html: str. The new translation_html string.
+        user_id: str. The id of the user who updates the suggestion.
+
+    Raises:
+        Exception. The suggestion is already accepted.
     """
     suggestion = get_suggestion_by_id(suggestion_id)
 
-    # Clean the translation HTML if not a list of strings.
-    suggestion.change.translation_html = (
-        html_cleaner.clean(translation_html)
-        if isinstance(translation_html, str)
-        else translation_html
-    )
-    suggestion.edited_by_reviewer = True
-    suggestion.pre_update_validate(suggestion.change)
-    _update_suggestion(suggestion)
+    if suggestion.status == suggestion_models.STATUS_ACCEPTED:
+        raise Exception(
+            'The suggestion with id %s has been already accepted'
+            % (suggestion_id)
+        )
+
+    if suggestion.author_id != user_id:
+        suggestion.edited_by_reviewer = True
+
+    if suggestion.status == suggestion_models.STATUS_REJECTED:
+        change_dict = suggestion.change.to_dict()
+        change_dict['translation_html'] = (
+            html_cleaner.clean(translation_html)
+            if isinstance(translation_html, str)
+            else translation_html
+        )
+        create_suggestion(
```

Comment:
Is this correct? It seems to me that the code can get to this point even if the suggestion author is not the same as the reviewer, so not sure why we're setting edited_after_rejecting to true.